grpc/codec.go

```diff
@@ -0,0 +1,41 @@
+package grpc
+
+import (
+	"io"
+
+	"google.golang.org/grpc/encoding"
+)
+
+// CodecName is the name registered for the proto compressor.
+const CodecName = "FiberCodec"
+
+// FiberCodec is a custom codec to prevent marshaling and unmarshalling
+// when unnecessary, base on the inputs
+type FiberCodec struct{}
+
+// Marshal will attempt to pass the request directly if it is a byte slice,
+// otherwise unmarshal the request proto using the default implementation
+func (FiberCodec) Marshal(v interface{}) ([]byte, error) {
+	b, ok := v.([]byte)
+	if ok {
+		return b, nil
+	}
+	defaultCodec := encoding.GetCodec("proto")
```

Comment:
Minor performance improvement, can we store the defaultCodec as the `FiberCodec`'s field?